### ISSUE TITLE
FeatureFinderIdentification parameter "detect:peak_width" affects feature width

### ISSUE THREAD

**Issue by hendrikweisser** [OpenMS/OpenMS#1935]
In FeatureFinderIdentification, the parameter "detect:peak_width" defines the width of the Gaussian kernel used for smoothing extracted ion chromatograms (in PeakPickerMRM). The value of this parameter has an effect on the width (RT span) of the features (elution peaks) that are detected and reported - higher "peak_width" leads to wider features. It would be better if the beginning and end of a feature was only determined by the data, not dependent on a detection parameter.

This problem may affect OpenSWATH more generally, i.e. not just on MS1 data.

**Comment by hroest** [comment]
yes, this is somewhat a relict of old times. One can use -1 for this parameter (this is what I recommend generally). However, we should also change the default.

also make sure you check the "method" parameter which either uses the original chromatogram or the smoothed chromatogram to pick peaks.

**Comment by hendrikweisser** [comment]
FFId uses -1 for the parameter "TransitionGroupPicker:PeakPickerMRM:peak_width". The "detect:peak_width" parameter that I refer to above is passed to OpenSWATH's "TransitionGroupPicker:PeakPickerMRM:gauss_width" parameter. (I use the Gauss filter, not Savitzky-Golay, because I can't guarantee that my data is uniformly spaced - although I don't know what difference it would make in practice.)

"TransitionGroupPicker:PeakPickerMRM:method" is set to "corrected", which I think is the recommended value?

**Comment by hroest** [comment]
yes, that makes a lot of sense and sounds good to me. I dont think we can get rid of the gauss_width parameter simply because that is how smoothing works and we dont know in advance what will be right for your chromatography.

But I agree, it would be better if this were determined by the data...

**Comment by hroest** [comment]
possibly fixed in #1953

**Comment by hendrikweisser** [comment]
#1953 just changes the default values of the parameters, so it doesn't solve this issue. I think in order to address this, the peak boundaries would have to be determined based on the original data, not the smoothed version.

**Comment by hroest** [comment]
ok, I see. however, all other workflows also require that you apply some smoothing before peak picking and then you also have to set the gaussian width yourself. However, it would be nice if this were done automatically

do we have code in OpenMS that would automatically detect the peak width? I think we do but the question is whether it would work.

**Comment by hendrikweisser** [comment]
> however, all other workflows also require that you apply some smoothing before peak picking and then you also have to set the gaussian width yourself. However, it would be nice if this were done automatically

Yes, I think we would probably need smoothing in any case to aid in detecting peaks, in order to ignore fluctuations in the intensity profile. Having a parameter that controls the smoothing is not the problem, although it would of course be nice if that could be set automatically. The problem is that the features themselves change depending on the value of the parameter (not so surprising if you think about the process, but nevertheless not really obvious and a bit annoying).

I think one way to go would be to detect peaks as before based on the smoothed data, but then add a refinement step that re-determines the peak boundaries based on the original data.

**Comment by hroest** [comment]
> I think one way to go would be to detect peaks as before based on the smoothed data, but then add a refinement step that re-determines the peak boundaries based on the original data.

This is also difficult as the original, raw data may often be very noisy (think at the very end of the peak where you are basically close to the noise) and using this to determine peak boundaries may also go wrong. However, there is the option to set the peak picker to "legacy" mode which basically performs the peak picking on the raw data instead of the smoothed data. Maybe this works for you or we can introduce a new mode that would introduce the "refinement" step as you suggested

**Comment by hendrikweisser** [comment]
I have found an especially egregious case that highlights the problem with determining the feature boundaries on the smoothed data. Based on an assay with two transitions (for two isotopes of a peptide) a feature was detected that:
- spanned an RT range of 45 seconds
- encompassed 38 data points (= spectra) in the XIC
- had only two (!) non-zero data points for one transition, none (!) for the other transition
- got scored highly
(Are zero-intensity points considered when elution peaks are compared e.g. for the cross-correlation scores?)

So basically the intensity from two data points was smeared over a wide enough range by the smoothing, which then gave rise to a "ghost" feature. I have not looked thoroughly, but this doesn't seem to be a totally isolated case in my data.

**Comment by hendrikweisser** [comment]
I tried to solve the problem mentioned above (features being detected in largely zero-intensity regions) by modifying the elution peak detection in PeakPickerMRM. In the "corrected" method (which detects peaks in smoothed chromatograms), I added a step that would adjust the peak boundaries inwards as long as the raw intensity at the boundary was zero.

Unfortunately that didn't help. The sensitivity of the feature detection decreased and reproducibility (CV across replicates) didn't improve. :-(
Intuitively it still makes sense to me to adapt the peak detection in this way, but then presumably the peak grouping across transitions/chromatograms would also have to be changed.
